api/src/controllers/person.js

```diff
@@ -8,20 +8,45 @@ const getPerson = ({ with_lineage }) => ctx.bind(
     ? Person.v1.getWithLineage
     : Person.v1.get
 );
+const getPageByType = () => ctx.bind(Person.v1.getPage);
+
+const checkUserPermissions = async (req) => {
+  const userCtx = await auth.getUserCtx(req);
+  if (!auth.isOnlineOnly(userCtx) || !auth.hasAllPermissions(userCtx, 'can_view_contacts')) {
+    return Promise.reject({ code: 403, message: 'Insufficient privileges' });
+  }
+};
 
 module.exports = {
   v1: {
     get: serverUtils.doOrError(async (req, res) => {
-      const userCtx = await auth.getUserCtx(req);
-      if (!auth.isOnlineOnly(userCtx) || !auth.hasAllPermissions(userCtx, 'can_view_contacts')) {
-        return Promise.reject({ code: 403, message: 'Insufficient privileges' });
-      }
+      await checkUserPermissions(req);
       const { uuid } = req.params;
       const person = await getPerson(req.query)(Qualifier.byUuid(uuid));
       if (!person) {
         return serverUtils.error({ status: 404, message: 'Person not found' }, req, res);
       }
       return res.json(person);
-    })
-  }
+    }),
+    getPageByType: serverUtils.doOrError(async (req, res) => {
+      await checkUserPermissions(req);
+
+      try {
+        const personType  = Qualifier.byContactType(req.query.personType);
+        const limit = req.query.limit !== undefined ? Number(req.query.limit) : 100;
+        const skip = req.query.skip !== undefined ? Number(req.query.skip) : 0;
+
+        // TODO: change this when #9281 gets merged
+        const docs = await getPageByType()( personType, limit, skip );
+
+        return res.json(docs);
+      } catch (err) {
+        if (err instanceof InvalidArgumentError) {
```

Comment:
Can you just put this logic down in the `server-utils.error` function?  (Basically just make sure the `code` gets set to `400` when we have an InvalidArgumentError).

Then, we would not need this try-catch block up here in `person.js`...